keps/sig-node/127-user-namespaces/README.md

```diff
@@ -316,121 +326,106 @@ The picked range will be stored under a file named `userns` in the pod folder
-namespace, but they will map to _the same_ UIDs/GIDs in the host.
+We tried this before, but several limitations were hit. See the 
+[alternatives section](#dont-use-idmap-mounts-and-rely-chown-all-the-files-correctly)
+for more details on the limitations we hit.
 
-Using the same mapping allows for pods to share files and mitigates all the
-listed vulnerabilities (as the host is protected from the container). It is also
-a trivial next-step to take, given that we have phase 1 implemented: just return
-the same mapping if the pod has other volumes.
+##### Example without idmap mounts
 
-While these pods do not use a distinct user namespace mapping, they are still
-using a new user namespace object in the kernel (so they cannot join/attack
-other pods namespaces). Security-wise this is a middle layer between what we
-have today (no userns at all) and using a distinct UID/GID mapping for the user
-namespace.
+Now let's say the pod is using its volumes that were mounted using idmap mounts
+by the container runtime. All the mappings used (the idmap mounts and the pod
+userns) are:
 
-#### Phase 3: TBD
+| Host ID  | Container ID   | Length  |
+|----------|----------------|---------|
+| 65536    |      0         |  65536  |
 
-#### Unresolved
 
-Here is a list of considerations raised in PRs discussion that hasn't yet
-settle. This list is not exhaustive, we are just trying to put the things that
-we don't want to forget or want to highlight it. Some things that are obvious we
-need to tackle are not listed. Let us know if you think it is important to add
-something else to this list:
+Keep in mind the file `/var/lib/kubelet/.../volumes/configmap/foo` is created by
+the kubelet, the owner is host UID 0, **but** it is bind-mounted into
+`/vol/configmap/foo` **using an idmap mount** with the mapping aforementioned.
+
+To achieve this, the kubelet only needs to set the UID/GID mappings in the
+`Mount` CRI message.
 
-- We will start with mappings of 64K. Tim Hockin, however, has expressed
-  concerns. See more info on [this Github discussion](https://github.com/kubernetes/enhancements/pull/3065#discussion_r781676224)
-  SergeyKanzhelev [suggested a nice alternative](https://github.com/kubernetes/enhancements/pull/3065#discussion_r807408134),
-  to limit the number of pods so we guarantee enough spare host UIDs in case we
-  need them for the future. There is no final decision yet on how to handle this.
-  For now we will limit the number of pods, so the wide mapping is not
-  problematic, but [there are downsides to this too](https://github.com/kubernetes/enhancements/pull/3065#discussion_r812806223)
+If the pod wants to read who is the owner of file `/vol/configmap/foo`, now it
+will see the owner is root inside the container. This is due to the IDs
+transformations that the idmap mount does for us.
 
+In other words, we can make sure the pod can read files instead of chowning them
```

Comment:
"we can make sure the pod can read files"

I like that as the goal, but it isn't clear (see above) that we achieve that if the container runs as non-root

Reply:
Let's center the discussion on that there.